### PR TITLE
feat: OpenVINO acceleration for the bat embedding extractor

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

Runs the bat model's heavy embedding extractor (`birdnet-v24-embeddings`) on OpenVINO instead of ONNX Runtime, mirroring the existing BirdNET v2.4 and Perch v2 pattern with OV-first and ORT fallback. OpenVINO never makes the bat model fail to load: any gate decline, init, compile, or validation error falls back to ORT. The tiny bat classifier head always stays on ORT (too small to benefit).

The device gate is identical to BirdNET v2.4 (reuses `openVINOPlanFor`): `auto` picks GPU when available, else CPU; `backend=onnx` or no eligible device falls back to ORT. The bat pipeline reuses the global `BirdNET.Backend` / `OpenVINOPath` / `OpenVINODevice` settings, so there is no new user-facing config.

## Precision: FP32 on every device

The one model-specific deviation from BirdNET v2.4 (which uses f16 on CPU, f32 only on GPU) is that the bat embedding model is forced to FP32 on every device. Its embedding head overflows at f16 on genuine f16 hardware (the A76 CPU's f16 NEON and the Intel iGPU), corrupting the raw embedding the bat classifier consumes and flipping detections. At FP32, OpenVINO beats ONNX Runtime on both CPU and GPU, so this is a performance win on whatever device the gate selects.

## Changes

- `NewOpenVINOEmbeddingExtractor` (internal/inference/openvino.go): reads a model's embedding output port and validates its dimension against the consumer's expected embedding size, falling back to ORT on mismatch.
- `InputDim()` added to the `inference.CustomClassifier` interface so the bat head's input dimension gates and validates the OpenVINO embedding path.
- `tryBatOpenVINO` in a reordered `NewBat`: build the ORT head first to read its input dim, then try OpenVINO with ORT fallback.
- `buildBat` / `loadBat` split records the backend triplet, and the bat model is registered as an OpenVINO-capable secondary so a runtime backend or device change rebuilds it via the existing hot-reload path.
- `openVINOPrecisionFor` forces FP32 for the bat model on every device.

## Validation

OpenVINO f32 vs ONNX Runtime f32 embedding parity, max absolute error, on real audio:

| Target | device | max abs error |
|--------|--------|---------------|
| amd64 i7-1260P Iris Xe | GPU | 0.0047 |
| amd64 i7-1260P | CPU | 0.0022 |
| rpi5 (Cortex-A76) | CPU | 0.0025 |

All three sit roughly 1000x below the f16-overflow regime (~5.0), confirming the forced-f32 path reproduces the ORT embedding on every device.

## Test Plan

- [x] `go build` (default and `-tags openvino`)
- [x] `go test -race` for `internal/classifier` and `internal/inference`
- [x] `golangci-lint run` clean (default build)
- [x] New unit tests: bat f32 precision (CPU and GPU), OV-first/ORT-fallback gating, dim-validation guards, hot-reload registration
- [x] New env-gated `-tags openvino` parity test, run on amd64 (GPU and CPU) and arm64 (A76 CPU)

## Release Notes

* **New Features**
  * Added OpenVINO acceleration for bat embedding extraction, with automatic fallback to ONNX Runtime when OpenVINO isn’t available or can’t compile.

* **Improvements**
  * Bat model runtime settings (backend/device/precision) are now properly detected and recorded, so secondary model reloads update correctly when configuration changes.
  * Bat OpenVINO embedding now consistently uses required f32 precision for matching results.

### CODE REVIEW

File: internal/inference/openvino.go [hunk 1]
```diff
@@ -111,6 +111,111 @@ func (c *openvinoClassifier) Close() {
...
+// only the embedding output port (the embedding model's logits port is not compiled
+// in), so Predict returns the embedding rather than logits. The bat pipeline calls
+// PredictWithEmbeddings, not Predict; Predict exists to satisfy the Classifier
+// interface that EmbeddingExtractor embeds.
+func (e *openvinoEmbeddingExtractor) Predict(samples []float32) ([]float32, error) {
+	if e.c == nil {
+		return nil, ov.ErrOpenVINOUnavailable
+	}
+	return e.c.PredictRaw(samples)
+}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/inference/openvino.go]: Good catch, agreed. Fixed in 10b65a4f: Predict now returns an explicit error since this extractor compiles only the embedding port and cannot produce logits, so a caller using it through a generic Classifier reference fails loudly instead of receiving an embedding vector in place of logits. The bat pipeline only ever calls PredictWithEmbeddings.
